tiatoolbox/utils/image.py

```diff
@@ -20,17 +20,183 @@
+
+    if padding_size == 1:
+        padding = np.repeat(padding, 4)
+    elif padding_size == 2:
+        padding = np.tile(padding, 2)
+    else:
+        padding = np.array(padding)
+    return padding
+
+
+def make_bounds_size_positive(bounds):
+    """Make bounds have positive size and get horizontal/vertical flip flags.
+
+    Bounds with a negative size in either direction with have the
+    coordinates swapped (e.g. left and right or top and bottom swapped)
+    and a respective horizontal or vertical flip flag set in the
+    output to reflect the swaps which occurred.
+
+    Args:
+        bounds (:class:`numpy.ndarray`): Length 4 array of bounds.
+
+    Returns:
+        tuple: Three tuple containing positive bounds and flips:
+            - :class:`numpy.ndarray` - Positive bounds
+            - :py:obj:`bool` - Horizontal flip
+            - :py:obj:`bool` - Vertical flip
+    """
+    hflip, vflip = False, False
+    _, (width, height) = bounds2locsize(bounds)
+    if width >= 0 and height >= 0:
+        return bounds, hflip, vflip
+    l, t, r, b = bounds
+    if width < 0:
+        l, r = r, l
+        hflip = True
+    if height < 0:
+        t, b = b, t
+        vflip = True
+    bounds = np.array([l, t, r, b])
+    return (bounds, hflip, vflip)
+
+
+def crop_and_pad_edges(
+    bounds: Tuple[int, int, int, int],
+    max_dimensions: Tuple[int, int],
+    region: np.ndarray,
+    pad_mode: str = "constant",
+    pad_constant_values: Union[int, Tuple] = 0,
+) -> np.ndarray:
+    """Apply padding to areas of a region which are outside max dimensions.
+
+    Applies padding to areas of the image region which have coordinates
+    less than zero or above the width and height in `max_dimensions`.
+    Note that bounds and max_dimensions must given for the same image
+    pyramid level (or more generally resolution e.g. if interpolated
+    between levels or working in other units).
+
+    Note: This function is planned to be deprecated in the future when a
+    transition from OpenSlide to tifffile as a dependency is complete.
+    It is currently used to remove padding from OpenSlide regions before
+    applying custom padding via :func:`numpy.pad`. This allows the
+    behaviour when reading OpenSlide images to be consistent with other
+    formats.
+
+    Args:
+        bounds (tuple(int)): Bounds of the image region.
+        max_dimensions (tuple(int)): The maximum valid x and y
+            values of the bounds, i.e. the width and height of the
+            slide.
+        region (:class:`numpy.ndarray`): The image region to be cropped
+            and padded.
+        pad_mode (str): The pad mode to use, see :func:`numpy.pad`
+            for valid pad modes. Defaults to 'constant'.
+        pad_constant_values (int or tuple(int)): Constant value(s)
+            to use when padding. Only used with pad_mode constant.
+
+    Returns:
+        :class:`numpy.ndarray`: The cropped and padded image.
+    """
```

Comment:
Please add a new line at the end.

Reply:
Fixed in this commit https://github.com/TIA-Lab/tiatoolbox/pull/61/commits/5b0e0ecc1fcb9ce989d907483bef9a2b0b2c7aa5